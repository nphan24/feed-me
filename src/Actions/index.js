export const postRecipes = recipes => ({
  type: 'POST_RECIPES',
  recipes
});

export const replaceRecipes = recipes => ({
  type: 'REPLACE_RECIPES',
  recipes
});

export const addFavorite = recipe => ({
  type: 'ADD_FAVORITE',
  recipe
});

export const removeFavorite = id => ({
  type: 'REMOVE_FAVORITE',
  id
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const promptSignIn = error => ({
  type: 'PROMPT_SIGNIN',
  error
});