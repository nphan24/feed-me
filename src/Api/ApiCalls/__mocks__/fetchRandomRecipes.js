export const fetchRandomRecipes = jest.fn().mockImplementation(() => [
  {
    Ingredients:[
      'butter',
      'heavy cream',
      'salt',
      'ground black pepper',
      'russet potatoes',
      'parmigiano'
    ],
    id: 'Potatoes-Au-Gratin-2244864',
    image:
      'https://lh3.googleusercontent.com/1UaqgvAZrw346jUpat4-at6mkz6-Jhjymn8AkVY2Kwu48E0TXbuH9OnhsR_k5OF-5UV_MSxx6qbnFPnawzTtqQ=s90-c',
    name: 'Potatoes Au Gratin',
    source: 'https://www.onceuponachef.com/recipes/potatoes-au-gratin.html',
    totaltime: 75
  }
]);
