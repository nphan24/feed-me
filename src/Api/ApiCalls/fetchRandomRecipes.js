import { apiKey, applicationId } from './apiKey';
import { cleanRecipes } from 
  '../helper/cleanRecipes';
import { fetchRecipeSource } from './fetchRecipeSource';

export const fetchRandomRecipes = async (selected) => {
  try {
    let category = selected ? selected : '';
    // let category = 'chicken';
    console.log('select', category);
    const response = await fetch(`http://api.yummly.com/v1/api/recipes?_app_id=${applicationId}&_app_key=${apiKey}&q=${category}&maxResult=30&start=00`);
    const data = await response.json();
    console.log(data)
    const recipesWithoutSource = await cleanRecipes(data.matches);
    const recipesWithSource = await fetchRecipeSource(recipesWithoutSource);
    return recipesWithSource;
  } catch (error) {
    throw new error(error.message);
  }
};

