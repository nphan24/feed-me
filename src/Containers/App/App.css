@import url('https://fonts.googleapis.com/css?family=PT+Serif:400,700');

.App-header {
  background-color: #2f97ef;
  height: 120px;
  padding: 20px;
  color: white;
}
.App-title {
  color: white;
  font-size: 2.5em;
  text-decoration: none;
  visibility: center;
  font-family: 'PT Serif',
  serif;
}

.top-icon {
  width: 40px;
  height: 40px;
  filter: invert(100);
  vertical-align: center;
  margin-right: 5px;
}

.App-title:hover {
  color: #E1DCDC;
}

.view-button {
  color: white;
  text-decoration: none;
  margin: 5px;
}

.view-favorites-button {
  color: white;
  text-decoration: none;
  margin: 5px;
  text-align: right;
  display: block;
  font-size: 1.5em;
  font-family: 'PT Serif',
  serif;
}

.view-favorites-button:hover {
  color: #E1DCDC;
}

.logout-button {
  border: none;
  font-size: 1.5em;
  font-family: 'PT Serif',
  serif;
  color: white;
  background-color: #2f97ef;
  float: right;
}

.logout-button:hover {
  color: #E1DCDC;
}