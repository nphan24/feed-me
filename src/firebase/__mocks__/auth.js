export const login = jest.fn().mockImplementation(() => [{
  email: 'dog@gmail.com', 
  password: 'dogdog', 
  uid: 24}
]);