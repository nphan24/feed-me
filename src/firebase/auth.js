import { auth } from './firebase';

//sign up
export const signUp = (email, password) => 
  auth.createUserWithEmailAndPassword(email, password);

  //login
export const login = (email, password) =>
  auth.signInWithEmailAndPassword(email, password);

  //signout
export const doSignOut = () => auth.signOut();