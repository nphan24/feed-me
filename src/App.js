import React, { Component } from 'react';
import { fetchData } from './Api/ApiCalls/fetchData';
import logo from './logo.svg';
import './App.css';

class App extends Component {

  async componentDidMount() {
    const mealData = await fetchData();
    console.log('meal data', mealData);
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">My Husband Should Cook This To Feed-Me</h1>
        </header>
      </div>
    );
  }
}

export default App;
