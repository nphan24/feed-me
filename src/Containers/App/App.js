import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchRandomRecipes } from '../../Api/ApiCalls/fetchRandomRecipes';
import CardContainer from '../CardContainer/CardContainer';
import DropDown from '../DropDown/DropDown';
import * as Actions from '../../Actions';
import { NavLink, withRouter } from 'react-router-dom';
import Nav from '../../Components/Nav/Nav';
import { auth } from '../../firebase';
import PropTypes from 'prop-types';
import './App.css';

const topIcon = require('../../assets/top-icon.svg');
export class App extends Component {
  async componentDidMount() {
    const recipes = await fetchRandomRecipes();
    this.props.postRecipes(recipes);
  }

  handleLogout = () => {
    auth.doSignOut();
    this.props.history.push('/');
    this.props.logoutUser();
  }

  handleFavorites = () => {
    const error = 'Please sign in to view Favorites';
    this.props.promptSignIn(error);
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <NavLink 
            to="/" 
            className="App-title">
            <img src={topIcon} className="top-icon"/>
            Feed-Me
          </NavLink>
          {this.props.user.email &&
          <NavLink 
            className="view-favorites-button" to="/favorites">
            View Favorites
          </NavLink>
          }
          {this.props.user.email && 
          <button  className="logout-button" 
            onClick={this.handleLogout}>
            Logout
          </button>}
          {!this.props.user.email &&
          <button 
            onClick={this.handleFavorites}
            className='favorites-button-not-signedin'>
            View Favorites</button>}
          {!this.props.user.email && <Nav />}
          <DropDown />
        </header>
        {this.props.error && 
          !this.props.user.email && 
          <p className='error-message'>{this.props.error}</p>}
        <CardContainer />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user,
  error: state.error
});

export const mapDispatchToProps = dispatch => ({
  postRecipes: recipes => dispatch(Actions.postRecipes(recipes)),
  logoutUser: user => dispatch(Actions.logoutUser(user)),
  promptSignIn: error => dispatch(Actions.promptSignIn(error))
});

App.propTypes = {
  postRecipes: PropTypes.func,
  history: PropTypes.object,
  logoutUser: PropTypes.func,
  user: PropTypes.object,
  promptSignIn: PropTypes.func,
  error: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.object
  ])
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
