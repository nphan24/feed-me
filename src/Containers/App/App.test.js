import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import * as mock from '../../mockData/mockData';
import * as Actions from '../../Actions';
import { fetchRandomRecipes } from '../../Api/ApiCalls/fetchRandomRecipes';
import * as firebase from '../../firebase/auth';
jest.mock('../../Api/ApiCalls/fetchRandomRecipes');

describe('App', () => {
  let wrapper;
  let mockpostRecipes = jest.fn();
  let mocklogoutUser = jest.fn();
  let mockUser = {email:'dog.gmail.com', uid: 24};
  let mockHistory = {push: jest.fn()};

  beforeEach (() => {
    wrapper = shallow(<App 
      postRecipes={mockpostRecipes} 
      logoutUser={mocklogoutUser}
      user={mockUser}
      history={mockHistory}
    />, {
      disableLifecycleMethods: false
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchRandomRecipes', () => {
    expect(fetchRandomRecipes).toHaveBeenCalled();
  });

  it('should call postRecipes on componentDidMount', () => {
    expect(mockpostRecipes).toHaveBeenCalledWith(mock.recipes);
  });

  it('should call doSignOut on handleLogout', () => {
    firebase.doSignOut = jest.fn();
    wrapper.instance().handleLogout();
    expect(firebase.doSignOut).toHaveBeenCalled();
  });

  it('should call history.push with the correct params', () => {
    wrapper.instance().handleLogout();
    expect(mockHistory.push).toHaveBeenCalledWith('/');
  });

  it('should call logoutUser on handleLogout', () => {
    expect(mocklogoutUser).toHaveBeenCalled();
  });
});

describe('mapStateToProps', () => {
  const state = {user: {email:'dog@gmail.com', uid: 24}};
  const expected = { email: 'dog@gmail.com', uid: 24 };
  const mapped = mapStateToProps(state);
  expect(mapped.user).toEqual(expected);
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with the correct params in postRecipes', () => {
    const mockDispatch = jest.fn();
    const recipe = mock.mockRecipe;
    const expected = Actions.postRecipes(recipe);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.postRecipes(recipe);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});
