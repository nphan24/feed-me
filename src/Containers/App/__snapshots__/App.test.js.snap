// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    history={
        Object {
            "push": [Function],
          }
    }
    logoutUser={[Function]}
    postRecipes={[Function]}
    user={
        Object {
            "email": "dog.gmail.com",
            "uid": 24,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="App-title"
                    to="/"
          >
                    <img
                              className="top-icon"
                              src="top-icon.svg"
                    />
                    Feed-Me
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="view-favorites-button"
                    to="/favorites"
          >
                    View Favorites
          </NavLink>
          <button
                    className="logout-button"
                    onClick={[Function]}
          >
                    Logout
          </button>
          <Connect(DropDown) />
</header>,
        <withRouter(Connect(Component)) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="App-title"
              to="/"
>
              <img
                            className="top-icon"
                            src="top-icon.svg"
              />
              Feed-Me
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="view-favorites-button"
              to="/favorites"
>
              View Favorites
</NavLink>,
            <button
              className="logout-button"
              onClick={[Function]}
>
              Logout
</button>,
            false,
            <Connect(DropDown) />,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                <img
                  className="top-icon"
                  src="top-icon.svg"
/>,
                "Feed-Me",
              ],
              "className": "App-title",
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "top-icon",
                  "src": "top-icon.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "Feed-Me",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "View Favorites",
              "className": "view-favorites-button",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "View Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
              "className": "logout-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logout",
            "type": "button",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="App-title"
                        to="/"
            >
                        <img
                                    className="top-icon"
                                    src="top-icon.svg"
                        />
                        Feed-Me
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="view-favorites-button"
                        to="/favorites"
            >
                        View Favorites
            </NavLink>
            <button
                        className="logout-button"
                        onClick={[Function]}
            >
                        Logout
            </button>
            <Connect(DropDown) />
</header>,
          <withRouter(Connect(Component)) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="App-title"
                to="/"
>
                <img
                                className="top-icon"
                                src="top-icon.svg"
                />
                Feed-Me
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="view-favorites-button"
                to="/favorites"
>
                View Favorites
</NavLink>,
              <button
                className="logout-button"
                onClick={[Function]}
>
                Logout
</button>,
              false,
              <Connect(DropDown) />,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": Array [
                  <img
                    className="top-icon"
                    src="top-icon.svg"
/>,
                  "Feed-Me",
                ],
                "className": "App-title",
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "top-icon",
                    "src": "top-icon.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "Feed-Me",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "View Favorites",
                "className": "view-favorites-button",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "View Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
                "className": "logout-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Logout",
              "type": "button",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;
